conda_libmamba_solver/repoquery.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (C) 2019, QuantStack
+# Copyright (C) 2022 Anaconda, Inc
+# Copyright (C) 2023 conda
+# SPDX-License-Identifier: BSD-3-Clause
+import argparse
+import json
+import sys
+
+from conda.base.context import context
+from conda.cli import conda_argparse
+from conda.common.io import Spinner
+from conda.core.prefix_data import PrefixData
+from libmambapy import QueryFormat
+
+from .index import LibMambaIndexHelper
+
+
+def configure_parser(parser: argparse.ArgumentParser):
+    package_cmds = argparse.ArgumentParser(add_help=False)
+    package_cmds.add_argument("package_query", help="the target package")
+    package_grp = package_cmds.add_argument_group("Subcommand options")
+    package_grp.add_argument(
+        "-i",
+        "--installed",
+        action="store_true",
+        default=True,
+        help=argparse.SUPPRESS,
+    )
+
+    package_grp.add_argument(
+        "-p",
+        "--platform",
+        default=context.subdir,
+        help="Platform/subdir to search packages for. Defaults to current platform.",
+    )
+    package_grp.add_argument(
+        "--no-installed", action="store_true", help="Do not search currently installed packages."
+    )
+    package_grp.add_argument(
+        "--pretty", action="store_true", help="Prettier output with more details."
+    )
+
+    package_grp.add_argument(
+        "-a",
+        "--all-channels",
+        action="store_true",
+        help="Look at all channels (for depends / whoneeds)",
+    )
+
+    view_cmds = argparse.ArgumentParser(add_help=False)
+    view_grp = view_cmds.add_argument_group("Dependency options")
+    view_grp.add_argument(
+        "-t", "--tree", action="store_true", help="Show dependencies in a tree-like format"
+    )
+    view_grp.add_argument("--recursive", action="store_true", help="Show dependencies recursively")
+
+    subparser = parser.add_subparsers(dest="subcmd")
+
+    c1 = subparser.add_parser(
+        "whoneeds",
+        help="shows packages that depend on this package",
+        parents=[package_cmds, view_cmds],
+    )
+
+    c2 = subparser.add_parser(
+        "depends",
+        help="shows dependencies of this package",
+        parents=[package_cmds, view_cmds],
+    )
+
+    c3 = subparser.add_parser(
+        "search",
+        help="shows all available package versions",
+        parents=[package_cmds],
+    )
+
+    for cmd in (c1, c2, c3):
+        conda_argparse.add_parser_channels(cmd)
+        conda_argparse.add_parser_networking(cmd)
+        conda_argparse.add_parser_known(cmd)
+        conda_argparse.add_parser_json(cmd)
+
+
+def repoquery(args):
+    if not args.subcmd:
+        print("repoquery needs a subcommand (search, depends or whoneeds)", file=sys.stderr)
+        print("eg:", file=sys.stderr)
+        print("    $ conda repoquery search python\n", file=sys.stderr)
+        sys.exit(1)
```

Comment:
Please don't `sys.exit` here, since it'll prevent `post-commands` hooks from firing by killing the Python process. Just return the exit value and it'll be bubbled up.

```suggestion
        print("repoquery needs a subcommand (search, depends or whoneeds), e.g.:", file=sys.stderr)
        print("    conda repoquery search python\n", file=sys.stderr)
        return 1
```

Reply:
Oh, it would be cool to ask the subparser to print a standardized help blob based on how it was configured instead of manually doing this here, but it's a nice to have